src/main/java/com/google/classes/User.java

```diff
@@ -0,0 +1,171 @@
+package com.java.sps.classes;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import com.google.appengine.api.blobstore.BlobKey;
+
+/*
+ * This class represents a single user of the chatroom platform.
+ *
+ */
+public final class User{
+    private final static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private String email;
+    private String username;
+    private BlobKey displayImageKey;
+    private final College college;
+    private final Key key;
+    private boolean updated; 
+    private float sentiment;
+
+    private User(String email, String username, College college, boolean isProf){
+        this.email = email;
+        this.username = username;
+        this.college = college;
+        this.isProf = isProf;
+    }
+
+    /*
+     * Returns the email of the User object
+     * 
+     */
+    public String getEmail(){
+        return this.email;
+    }
+
+        /*
+     * Returns the College object of the User object
+     *
+     */
+    public College getCollege(){
+        return this.college;
+    }
+
+    /*
+     * Returns the username of the User object
+     * 
+     */
+    public String getName(){
+        return this.username;
+    }
+
+    /*
+     * Returns the BlobKey of the displayImage of a User object
+     */
+    public BlobKey getDisplayImageKey(){
+        return this.key;
+    }
+
+    /*
+     * Updates the email of a User object
+     */
+    public void updateEmail(String newEmail){
+        this.email = newEmail;
+        if (key == null) return;
+        Entity user = datastore.get(this.key);
+        user.setProperty("email", this.email);
+        datastore.put(user);
+    }
+
+    /*
+     * Updates the username of a User object
+     */
+    public void updateName(String newUsername){
+        this.username = newUsername;
+        if (key == null) return;
+        Entity user = datastore.get(this.key);
+        user.setProperty("username", this.username);
+        datastore.put(user);
+    }
+
+    /*
+     * Updates the BlobKey of the displayImage of a User object
+     */
+    public void updateDisplayImageKey(BlobKey newKey){
+        this.displayImageKey = newKey;
+        if (key == null) return;
+        Entity user = datastore.get(this.key);
+        user.setProperty("displayImageKey", this.displayImageKey);
+        datastore.put(user);
+    }
+
+    public void updateSentiment(int newSentiment){
```

Comment:
Should be a float, right? Unless we go with the running total and number of times updated

Reply:
Yeah this should be a `float`. Will update it. I am thinking of internal counter that stores the number of times the sentiment value has been updated and then will calculate the new average.
For example:
`sentiment = (numOfUpdates*sentiment + newSentiment)/(numOfUpdates + 1);`
`numOfUpdates++;`